unix-domain-socket/src/main/scala/akka/stream/alpakka/unixdomainsocket/scaladsl/UnixDomainSocket.scala

```diff
@@ -0,0 +1,526 @@
+                  if (bytes.nonEmpty) {
+                    val (left, right) = bytes.splitAt(sendBufferSize)
+                    splitToBufferSize(right, acc :+ left)
+                  } else {
+                    acc
+                  }
+                splitToBufferSize(bytes, List.empty).toIterator
+              }
+            }
+            .mapAsync(1) { bytes =>
+              // Note - it is an error to get here and not have an AvailableSendContext
+              val sent = Promise[Done]
+              val sendBuffer = sendReceiveContext.send.buffer
+              sendBuffer.clear()
+              val copied = bytes.copyToBuffer(sendBuffer)
+              sendBuffer.flip()
+              require(copied == bytes.size) // It is an error to exceed our buffer size given the above expand
+              sendReceiveContext.send = SendRequested(sendBuffer, sent)
+              sel.wakeup()
+              sent.future.map(_ => bytes)
+            }
+            .watchTermination() {
+              case (mat, done) =>
+                done.onComplete { _ =>
+                  sendReceiveContext.send = CloseRequested
+                  sel.wakeup()
+                }
+                (mat, done)
+            }
+            .runWith(Sink.ignore)
+        }
+    (sendReceiveContext, Flow.fromSinkAndSourceCoupled(sendSink, Source.fromFutureSource(receiveSource)))
+  }
+
+  /*
+   * A flow that requires the downstream to complete for it to complete i.e. it will
+   * keep going if the upstream completes.
+   */
+  private class HalfCloseFlow extends GraphStage[FlowShape[ByteString, ByteString]] {
+    private val in = Inlet[ByteString]("HalfCloseFlow.in")
+    private val out = Outlet[ByteString]("HalfCloseFlow.out")
+
+    override val shape: FlowShape[ByteString, ByteString] = FlowShape.of(in, out)
+
+    override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+      new GraphStageLogic(shape) {
+
+        override def preStart(): Unit =
+          setKeepGoing(true) // At a minimum, the downstream completion will stop this stage.
+
+        setHandler(in, new InHandler {
+          override def onPush(): Unit =
+            push(out, grab(in))
+
+          override def onUpstreamFinish(): Unit =
+            ()
+        })
+
+        setHandler(out, new OutHandler {
+          override def onPull(): Unit =
+            if (!isClosed(in) && !hasBeenPulled(in)) pull(in)
+        })
+      }
+  }
+}
+
+/**
+ * Provides Unix Domain Socket functionality to Akka Streams with an interface similar to Akka's Tcp class.
+ */
+final class UnixDomainSocket(system: ExtendedActorSystem) extends Extension {
+
+  import UnixDomainSocket._
+
+  private implicit val materializer: ActorMaterializer = ActorMaterializer()(system)
+  import system.dispatcher
+
+  private val sel = NativeSelectorProvider.getInstance.openSelector
+
+  private val ioThread = new Thread((() => nioEventLoop(sel)): Runnable, "unix-domain-socket-io")
```

Comment:
CI failed here. Since we still test and release for Scala 2.11 we can not use a lambda expression when SAM is expected. You will have to create the Runnable explicitly.